lib/libc/stdlib/malloc/mrs/mrs.c

```diff
@@ -1451,16 +1456,25 @@ mrs_free(void *ptr)
 void *
 mrs_mallocx(size_t size, int flags)
 {
+	size_t align = MALLOCX_ALIGN_GET(flags);
+	void *ret;
+
 	if (!quarantining)
 		return (REAL(mallocx)(size, flags));
 
-	/*
-	 * This API supports optional zeroing.  Since no one really uses
-	 * it, we don't care about performance and preserve the ABI by
-	 * unconditionally calling calloc.  (We should always be zeroing
-	 * for correctness anyway.)
-	 */
-	return (mrs_calloc(1, size));
+	if (align < CAPREVOKE_BITMAP_ALIGNMENT)
+		ret = mrs_malloc(size);
+	else if (mrs_posix_memalign(&ret, size, align) != 0)
+		ret = NULL;
+	if (ret == NULL)
```

Comment:
```suggestion

	if (ret == NULL)
```

Reply:
Though could also guard the code below with != NULL instead and only have one return